dm/pkg/checker/table_structure.go

```diff
@@ -79,68 +103,67 @@ func (c *TablesChecker) Check(ctx context.Context) *Result {
 		Name:  c.Name(),
 		Desc:  "check compatibility of table structure",
 		State: StateSuccess,
-		Extra: fmt.Sprintf("address of db instance - %s:%d", c.dbinfo.Host, c.dbinfo.Port),
 	}
 
-	var (
-		err        error
-		options    = make(map[string][]*incompatibilityOption)
-		statements = make(map[string]string)
-	)
-	for schema, tables := range c.tables {
-		if len(tables) == 0 {
-			tables, err = dbutil.GetTables(ctx, c.db, schema)
-			if err != nil {
-				markCheckError(r, err)
-				return r
-			}
-		}
+	startTime := time.Now()
 
-		for _, table := range tables {
-			tableName := dbutil.TableName(schema, table)
-			statement, err := dbutil.GetCreateTableSQL(ctx, c.db, schema, table)
-			if err != nil {
-				// continue if table was deleted when checking
-				if isMySQLError(err, mysql.ErrNoSuchTable) {
-					continue
-				}
-				markCheckError(r, err)
-				return r
-			}
+	for sourceID := range c.tableMap {
+		c.firstSourceID = sourceID
+		break
+	}
 
-			opts := c.checkCreateSQL(ctx, statement)
-			if len(opts) > 0 {
-				options[tableName] = opts
-				statements[tableName] = statement
-			}
+	checkCtx, cancel := context.WithCancel(ctx)
```

Comment:
Seems this WithCancel is no need because you didn't call cancel() in this function.